tests/test_delay.py

```diff
@@ -0,0 +1,245 @@
+import pytest
+from jobfunnel.tools.tools import config_factory
+from jobfunnel.tools.delay import delay_alg
+
+# Define mock data for this test module
+
+linear_delay = [0, 0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6, 1.8]
+sigmoid_delay = [0, 0.263, 0.284, 0.307,
+                 0.332, 0.358, 0.386, 0.417, 0.449, 0.485]
+constant_delay = [0, 8.6, 8.8, 9.0, 9.2, 9.4, 9.6, 9.8, 10.0, 10.0]
+random_delay = [0, 5, 5, 5, 5, 5, 5, 5, 5, 5]
+job_list = ['job1', 'job2', 'job3', 'job4', 'job5',
+            'job6', 'job7', 'job8', 'job9', 'job10']
+
+
+# mock random.uniform to get constant values
+
+def mock_rand_uniform(a, b):
+    return 5
+
+
+# test linear, constant and sigmoid delay a min_delay greater than the delay
+
+def test_delay_alg_linear_min_delay_greater_than_delay(configure_options):
```

Comment:
It's great that we are adding tests for these.

One thing that I'd like to see in this case, since we are doing so much of the same setup and now just changing parameters is for you to write a parameterized test.

In this case, we can do this:

```
@pytest.mark.parametrize(
    'function, min_delay, exp_result', 
    [
        ('linear', 15, linear_delay),
        ('sigmoid', 15,  sigmoid_delay),
)
@pytest.mark.parametrize('is_random', [True, False])
@pytest.mark.parametrize('delay', [0, -2])
@pytest.mark.parametrize('min_delay', [0, -2])
def test_delay_alg(is_random, function, delay, min_delay, exp_result, configure_options) 
    ... #(test here and use the passed args )

```
Note that you could express each test method below where we pass all the arguments or we could mark seperately things like is_random and it will test all possible combinations

more details: https://docs.pytest.org/en/latest/parametrize.html

Doing this will cut down on the liklihood of copypasta in the future as well. Basically most of this file can be reduced to one test case with it :)

Reply:
Looks really neat! Will be implementing later in the evening :)

Reply:
Has been fixed on 45d3c402b7a6e424aadae67e72bdf2af95b84901 :)